### PR TITLE
[DotNetCore] Implementing `dotnet migrate` support inside IDE

### PR DESCRIPTION
**Pull request by DavidKarlas**
image

image

image

image

### CODE REVIEW

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		protected override async void Run (object dataItem)
+		{
+			using (var monitor = IdeApp.Workbench.ProgressMonitors.GetToolOutputProgressMonitor (true)) {
+				try {
+					var project = IdeApp.ProjectOperations.CurrentSelectedProject as UnknownProject;
+					var solution = IdeApp.ProjectOperations.CurrentSelectedSolution;
+					FilePath projectFile;
+					string migrationDirectory;
+					if (project == null) {
+						migrationDirectory = Path.GetDirectoryName (solution.FileName);
```
Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs]: Just wondering if we should show a dialog when migrating the solution and allow the user to select the projects they want to migrate. That would be similar to what Visual Studio does.

Comment (slluis) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs]: I agree

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+					} else {
+						projectFile = project.FileName;
+						migrationDirectory = Path.GetDirectoryName (projectFile);
+					}
+					var dotnetPath = new DotNetCorePath ();
+					if (dotnetPath.IsMissing) {
+						monitor.ReportError (GettextCatalog.GetString (".NET Core is not installed"));
+						return;
+					}
+					monitor.BeginTask (GettextCatalog.GetString ("Migrating"), 1);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs]: I think this should have an ellipsis (`…`) in the task title.

**Comment by DavidKarlas** [comment]
Project picker dialog:
image

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+				} finally {
+					dlg.Destroy ();
+					dlg.Dispose ();
+				}
+			} else {
+				selectedProjects = new UnknownProject [] { project };
+			}
+			using (var monitor = IdeApp.Workbench.ProgressMonitors.GetToolOutputProgressMonitor (true)) {
+				try {
+					monitor.BeginTask (GettextCatalog.GetString ("Migrating…"), 1);
```
Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs]: @vancura - Is that a real "..." will be happy 👍

Comment (vancura) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetMigrate.cs]: Yeah it is :) Thanks! :)

**Pull request merged**